api/internal/managers/installation/manager.go

```diff
@@ -104,12 +97,8 @@ func NewInstallationManager(opts ...InstallationManagerOption) *installationMana
 		manager.logger = logger.NewDiscardLogger()
 	}
 
-	if manager.installation == nil {
-		manager.installation = types.NewInstallation()
-	}
-
 	if manager.installationStore == nil {
-		manager.installationStore = NewMemoryStore(manager.installation)
+		manager.installationStore = NewMemoryStore(manager.rc, types.NewStatus())
```

Comment:
The status needs to be tied to the one in the "install" type i think, can't be a new one, otherwise it wouldn't update correctly there. That's why the installation object was passed down as a reference

Reply:
before this change, the `manager.installation` would be a pointer to the one on the `install` object here: https://github.com/replicatedhq/embedded-cluster/blob/7f687075fa235d9ec4a916f4f74b84bce1ec0698/api/controllers/install/controller.go#L130

so the changes reflect in the `install` object. with this change, it's creating a new status and changes wouldn't reflect in the `install` object.

Reply:
Perhaps we need to refactor out these pointers? It is hard to understand and has side effects.

Reply:
Yeah agreed